src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs

```diff
@@ -238,6 +238,27 @@ public ImageSource LargeIcon
             }
         }
 
+        /// <summary>
+        /// Indicates if the user can view the confidence score or the recent use icon
+        /// </summary>
+        public Boolean ViewConfidenceScoreRecentUse { get; set; }
+        /// <summary>
+        /// Rating of confidence
+        /// </summary>
+        public int ConfidenceScore { get; set; }
+        /// <summary>
+        /// Indicates if the Node is part of ML result per use
+        /// </summary>
+        public Boolean IsByUse { get; set; }
+        /// <summary>
+        /// Indicates if the Node is part of the ML result per recommendation
+        /// </summary>
+        public Boolean IsByRecommendation
+        {
+            get
+            { return !IsByUse; }
+        }
```

Comment:
This class is shared for incanvas search and node autocomplete search. How does your current changes affect the look for incanvas search results?

Also, to make the usage more appearant, maybe we can create a new class inherit from the current one?

Reply:
I have tested the incanvas search and the new properties of the NodeSearchElementViewModel don't impact its look since in its UI implementation (SidebarGridStyleDictionary.xaml MemberTemplate) is being handling by the property ViewConfidenceScoreRecentUse, by default false. so it will keep hide the new mapped controls at least that we want to display them, that's the case when the user select the ML method.

About create a new class based on the current one  it means replace it in the NodeAutoCompleteSearchViewModel.cs and also in the SearchViewModel.cs so that mean several changes, per my understanding keeping the 4 new properties in the current class is valid since the search element will need to support its new source from the ML.

Reply:
@jesusalvino My take is the 4 new properties do not apply to the base class but only search element for NodeAutoComplete, in this case, can we create a sub class with these additional properties?

Reply:
After reviewing your current implementation, I think this is good for now